[ADD] Pricer tags section added to PoS/pricing (#6539)
odoo/documentation: content/applications/sales/point_of_sale/pricing/digital_labels.rst

==============
Digital labels
==============

Digital labels allow you to display important product information like product prices and barcodes
on store shelves and to update them remotely from the backend, eliminating the need to print new
labels when product information changes.

.. image:: digital_labels/electronic-label.png
   :alt: electronic label from Pricer

.. note::
   Odoo uses digital labels from `Pricer <https://www.pricer.com/>`_.

Configuration
=============

Pricer setup
------------

#. `Get in touch with Pricer <https://www.pricer.com/contact>`_ to create and configure your Pricer
   account.
#. Create your stores: one pricer store equates to one physical store.
#. Create the following variables to allow sharing of information between Odoo and Pricer:

   - `itemId` : the internal ID of your products
   - `itemName` : the name of your products
   - `price` : the price of your products, taxes included

>    - `presentation`: the template name used for Pricer tags display

Reviewer: I've changed 3 to try and make this clearer. There's no example because itemID should = itemID x) but maybe it's clearer this way? (I've had to ask for an explanation bc I, too had a hard time understanding)

   - `currency` : the currency of your company

>    - `barcode` : the barcode associated with your products

Reviewer: Oh, good catch, I didn't notice it

>    .. warning::

Reviewer: You're absolutely right

      The names for these variables must be **identical** in your Pricer database.

#. Create a template used to display information on your digital tags.

   .. warning::
      Your template must be named `NORMAL`.

Once your account, stores, variables, and template are configured on Pricer, you can proceed to

> the Odoo setup.

Reviewer: your version sounded weird to me because "the" referred to "setup". But I changed it and it's probably much better

.. warning::
   The account associated with your Pricer store must have access to send API requests to Pricer.

Odoo setup
----------

As a pre-requisite, :ref:`activate <general/install>` the :guilabel:`POS Pricer` module *(technical

> name: pos_pricer)* to have all the required features to use Pricer electronic tags.

Reviewer: because I say it in a non-important way in my head hahaha

.. image:: digital_labels/pricer-module.png
   :alt: Installing POS Pricer module from Apps

Once the module is activated, configure your :ref:`pricer stores <pricer_tags/stores>` and associate
:ref:`Pricer tags <pricer_tags/tags>` with your products.

.. _pricer_tags/stores:

Pricer stores
~~~~~~~~~~~~~

Similarly to the configuration in Pricer, you need to create one pricer store per physical location.
To do so, go to :menuselection:`Point of Sale --> Configuration --> Pricer Stores`, click
:guilabel:`New`, and fill in the line with the required information:

- :guilabel:`Store Name`: you can put any name of your liking.
- :guilabel:`Pricer Tenant Name`: the name of your company account in Pricer, usually followed by
  `-country_code`. This value is provided by Pricer.
- :guilabel:`Pricer Login` : the login of your Pricer account.
- :guilabel:`Pricer Password` : the password of your Pricer account.
- :guilabel:`Pricer Store ID`: the ID of the related Pricer store as defined on your Pricer

>   database.

Reviewer 2: Same issue with colons and spaces here as in the previous bullet list

.. image:: digital_labels/pricer-stores-setup.png
   :alt: Configuring a Pricer Store

.. note::
   - The :guilabel:`Pricer Tags` column is updated automatically when a label is linked to a
     product.
   - The :guilabel:`Last Update` and :guilabel:`Last Update Status` columns are updated
     automatically when the tags are updated.

.. _pricer_tags/tags:

Pricer tags
~~~~~~~~~~~

For a label to display one specific product information, they both need to be associated with each

> other. To do so:

Reviewer 2: Maybe it's me, but I needed to read this sentence three times to understand. It might be better to really spell it out to people, eg.:

For a label to display specific product information, the label needs to be associated with the product.

Up to you :)

#. Open the product form by going to :menuselection:`Point of Sale --> Products --> Products` and
   clicking :guilabel:`New` or selecting an existing product.

>    .. warning::

Reviewer 2: I would put it as a "note", it doesn't seem even like an "important" to me, and even less like a "warning"

      If you are creating a new product, configure and save it before associating any Pricer tags.

#. Go to the :guilabel:`Sales` tab, scroll to the :guilabel:`Pricer` section, and select the
   corresponding :guilabel:`Pricer Store`.

   .. image:: digital_labels/pricer-product.png
      :alt: Linking Pricer tags to products
      :scale: 75%

> #. Copy the label's ID from the label itself or scan its barcode to fill in the :guilabel:`Pricer
>    tags ids` field.

Reviewer 2: ```suggestion
#. Fill in the :guilabel:`Pricer tags ids` field by copying the label's ID from the label itself or scanning its barcode.
```

I'd phrase it like this, so we start by telling people _what_ to do (fill in this field), and follow with _how_ to do it (scanning the code).

   .. note::
      Pricer tag IDs are composed of a letter followed by 16 digits.

.. tip::
   - We recommended using a barcode scanner to speed up the encoding process.
   - When setting up Pricer with Odoo for the first time, it is recommended that you configure only
     one product first. Before configuring more products, ensure you can display its information on
     a Pricer tag.

Now that you have a product associated with a Pricer tag, we can send its information to Pricer.

Practical application
---------------------

> Odoo automatically sends requests to Pricer to update and synchronize the tags every 12 hours if

Reviewer 2: Just a terminology comment: I believe that what we discuss here is technically more of a synchronization than an update (update being more of new things introduced to the same system, whereas synchronization/sync is data getting aligned between two systems. HOPE THIS MAKES SENSE LOL).

So I would refer to this as "synchronization" (or "sync") in later sentences, HOWEVER, I think it really depends on what terminology Pricer uses on their end. Anyway this is nitpicky, do with it what you feel is best ;)

you make any modifications to:

>    - A product name, price, barcode, or customer taxes
>    - The currency

Reviewer 2: ```suggestion
   - Product name, price, barcode, or customer taxes
   - Currency
```
I think we can skip the articles in a bullet list (unless you really want to keep them to differentiate between product specific info in bullet one, and company level currency in bullet two)

   - Associated Pricer store or Pricer tags

To force the update, activate the :ref:`developer mode <developer-mode>`. Then:

#. Go to :menuselection:`Point of Sale --> Configuration --> Pricer Store`.
#. Select the desired store(s).
#. Click :guilabel:`Update tags` to update all tags affected by changes to:

   - A product name, price, barcode, or customer taxes
   - The currency
   - Associated Pricer store or Pricer tags

> Alternatively, click :guilabel:`Update all tags` to force the update of every tag, regardless of the
> changes made.

Reviewer 2: ```suggestion
Alternatively, click :guilabel:`Update all tags` to force the update of every tag, regardless if changes were made or not. If no changes were made, the tag information will stay the same.
```

I would specify a little bit more what will happen (I suppose nothing will happen if the info was not updated in Odoo but we force update/synchronization). Just for those readers that are really scared of tech :)
